homeassistant/components/integration/sensor.py

```diff
@@ -293,9 +291,10 @@ async def async_added_to_hass(self) -> None:
             self._unit_of_measurement = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
 
         @callback
-        def calc_integration(event: Event[EventStateChangedData]) -> None:
+        def calc_integration(event: Event[EventStateReportedData]) -> None:
             """Handle the sensor state changes."""
-            old_state = event.data["old_state"]
+            old_last_reported = event.data.get("old_last_reported")
```

Comment:
This is not used

Reply:
It is used, see below.

Reply:
In line 330 you were replacing it, but OK, this is for when old_state is None.

Reply:
Yes, the code is maybe hard to follow, I can try to rewrite it to make it clearer. We should probably merge the refactoring PR first anyway.